Resources/public/js/views/fields/ez-country-editview.js

```diff
@@ -0,0 +1,161 @@
+            }
+        },
+
+        _getSelectionFilter: function () {
+            var container = this.get('container'),
+                selectedObjectArray = this._getSelectedTextValues(),
+                input = container.one('.ez-selection-filter-input'),
+                source = this._getSource(),
+                selected = [];
+
+            Y.Array.each(selectedObjectArray, function (selectedObject) {
+                selected.push(selectedObject.text);
+            });
+            return new Y.eZ.SelectionFilterView({
+                container: input.get('parentNode'),
+                inputNode: input,
+                listNode: this.get('container').one('.ez-selection-options'),
+                selected: selected,
+                source: source,
+                filter: true,
+                resultFilters: 'startsWith',
+                resultHighlighter: 'startsWith',
+                isMultiple: this.get('isMultiple'),
+                resultTextLocator: function (sourceElement) {
+                    return sourceElement.Name;
+                },
+                resultAttributesFormatter: function (sourceElement) {
+                    return {
+                        alpha2: sourceElement.Alpha2,
+                        text: sourceElement.Name,
+                    };
+                },
+            });
+        },
+
+        _addSelection: function (attributes) {
+            var values = this.get('values');
+
+            values.push({text:attributes.text , alpha2: attributes.alpha2});
+            this._set('values', values, {"added": attributes.text});
+        },
+
+        _removeSelection: function (text, node) {
+            var values = this.get('values');
+
+            values = Y.Array.reject(values, function (val) {
+                return (val.text === text);
+            });
+            this._selectionFilter.unselect(text);
+            this._set('values', values, {
+                "removed": text,
+                "node": node || null
+            });
+        },
+
+        _getSource: function () {
+            var countriesArray = [];
+
+            Y.Object.each(this.get('config'), function (country) {
+                countriesArray.push(country);
+            });
+            return countriesArray;
+        },
+
+        _getSelectedTextValues: function () {
+            var field = this.get('field'),
+                valueIndexes = [],
+                that =  this,
+                res = [];
+
+            if ( field && field.fieldValue ) {
+                valueIndexes = field.fieldValue;
+            }
+            Y.Array.each(valueIndexes, function (index) {
+                res.push({text: that._getCountryName(index), alpha2: index});
+            });
+            return res;
+        },
+
+        _variables: function () {
+            var def = this.get('fieldDefinition'),
+                selected = [];
+
+            Y.Array.each(this.get('values'), function(value) {
```

Comment:
in the case of country, the `selected` template variable should probably contain the *country object* (ie the name + alpha2 code) so basically you can write:

```js
return {
    "isRequired": def.isRequired,
    "selected": this.get('values'),
    "isMultiple": this.get('isMultiple'),
};
```